WooCommerce/Classes/Model/Note+Woo.swift

```diff
@@ -66,3 +66,34 @@ private extension Note {
         static let emptyStar    = "\u{2606}"  // Unicode White Star ☆
     }
 }
+
+extension Note {
+    var orderID: Int? {
+        return meta.identifier(forKey: .order)
+    }
+
+    func markOrderNoteAsReadIfNeeded() {
+        // we sync the order notification to see if it is the notification for the last order
+        // we compare this note order id with synced note order id
+        // if they are the same it means that the notification is for the last order
+        // and we can mark it as read
+        guard read == false else {
+            return
+        }
+        let action = NotificationAction.synchronizeNotification(noteID: noteID) { syncronizedNote, error in
+            guard let syncronizedNote = syncronizedNote else {
+                return
+            }
+            if let syncronizedNoteOrderID = syncronizedNote.orderID,
+               let currentNoteOrderID = self.orderID,
+               syncronizedNoteOrderID == currentNoteOrderID {
+                // mark as read
+                let syncAction = NotificationAction.updateReadStatus(noteID: noteID, read: true) { error in
+                    DDLogError("⛔️ Error marking single notification as read: \(error)")
+                }
+                ServiceLocator.stores.dispatch(syncAction)
+            }
+        }
+        ServiceLocator.stores.dispatch(action)
+    }
```

Comment:
While this works, We tend to avoid accessing Yosemite from networking classes. 

One pattern we use frequently is to create a `MarkOrderAsReadUseCase` type where we can properly pass dependencies, like the `stores`, and test it appropriately.

Additionally, having a completion closure or making it async and return a completion value would be nice for future debugging!

Reply:
Even though this is in extension for `Note` we try to avoid that?

 I suppose since I only need note and order identifier I will make a method that uses just that. 👍

Reply:
> MarkOrderAsReadUseCase

Could you explain that a bit more? That would be some type with just a method that marks the note as read or?

Reply:
something like this?
```
class MarkOrderAsReadUseCase {
    static func markOrderNoteAsReadIfNeeded(stores: StoresManager, noteID: Int64, orderID: Int) {
        let action = NotificationAction.synchronizeNotification(noteID: noteID) { syncronizedNote, error in
            guard let syncronizedNote = syncronizedNote else {
                return
            }
            if let syncronizedNoteOrderID = syncronizedNote.orderID,
               syncronizedNoteOrderID == orderID {
                // mark as read
                let syncAction = NotificationAction.updateReadStatus(noteID: noteID, read: true) { error in
                    DDLogError("⛔️ Error marking single notification as read: \(error)")
                }
                stores.dispatch(syncAction)
            }
        }
        stores.dispatch(action)
    }
}

```
I would add a completion block also.

Reply:
I guess it could be a struct, and you could make the methods async to not deal with completion closures.

And maybe you could use compile flags and separate internal implementations to allow this to be shared with the watch & notifications extensions later.